sed/io.py

```diff
@@ -75,89 +113,97 @@ def res_to_xarray(res, bin_names, bin_axes, metadata=None):
 
     for name in bin_names:
         try:
-            xres[name].attrs['unit'] = DEFAULT_UNITS[name]
+            xres[name].attrs["unit"] = DEFAULT_UNITS[name]
         except KeyError:
             pass
 
-    xres.attrs['units'] = 'counts'
-    xres.attrs['long_name'] = 'photoelectron counts'
+    xres.attrs["units"] = "counts"
+    xres.attrs["long_name"] = "photoelectron counts"
 
     if metadata is not None:
-        xres.attrs['metadata'] = metadata
+        xres.attrs["metadata"] = metadata
 
     return xres
 
-def xarray_to_h5(data, faddr, mode='w'):
-    """ Save xarray formatted data to hdf5
+
+def xarray_to_h5(data: xr.DataArray, faddr: str, mode: str = "w"):
```

Comment:
Do we foresee any other mode than writing? If not, I would simplify the API and remove the mode from here

Reply:
I think, "w" overwrites files per default. If you want e.g. to append, you could change this with this option, that was the idea...

Reply:
Yes, the "w" mode creates files from scratch. 
In my opinion, this should be only option. Giving too many choices to users might be not a good idea, because it increases the chance of breaking things and the complexity of the tests needed for those options.

Reply:
Sure, for this one cannot provide test for any possible case, but similarly not for any possible content of an xarray. I find it useful as an optional parameter, that typically anyways is omitted...